### ISSUE TITLE
7xxxxxxx patch codes are not working

### ISSUE THREAD

**Issue by rothor** [PCSX2/pcsx2#2862]
***PCSX2 version:***
1.5-930f07d

***Description of the issue:***
Currently the source code for parsing 7-patch codes expects the patch code to be in the format:
`7aaaaaaa 00t0vvvv`
where "aaaaaaa" is the address to modify, "vvvv" is the value to compare to, and "t" signals whether to do an and, or, or xor and whether it is 8-bit or 16-bit. You can see the list of patch codes [here](https://sites.google.com/site/ps2hackingconverting/PS2-Commands).

Currently, the source code expects `00t0vvvv`, when the actual specification for codebreaker 7+ is `000tvvvv`.

***Replicating the issue:***
You can use
`patch=1,EE,7aaaaaaa,extended,002000FF`
which does an '& 0xFF' on the byte at aaaaaaa, and the value is correctly unchanged. Whereas using
`patch=1,EE,7aaaaaaa,extended,000200FF`
(which is what the documentation for codebreaker 7+ specifies) causes an incorrect result.

***Solution:***
This can be fixed in file "pcsx2/Patch_Memory.cpp" at line 264 by changing the `0x00F00000 == 0x00100000` to `0x000F0000 == 0x00010000`, and then the same at lines 259, 269, 274, 279, and 284.

***Some thoughts:***
There isn't any documentation for what patch codes are supported in PCSX2, or what cheat-device format to use. The link I provided is the only documentation that someone has for writing patch codes. The source code should match the format for codebreaker 7+, which is what it currently emulates.

**Comment by RhythmLady** [comment]
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ

**Comment by Mrlinkwii** [comment]
Is this still an issue ?

**Comment by rothor** [comment]
Yes it is, the 7-code patches are still not working. I just tested on the current nightly build (pcsx2-v1.7.3735-windows-64bit-SSE4-Qt), only now I'm not able to replicate my previous results on this current build. The behavior is still wrong but different than before.

Now the 7-code patch is behaving exactly like a 2-code patch (a 32-bit write). For example, the patch
`patch=1,EE,7aaaaaaa,extended,vvvvvvvv`
is now just writing the value of 'v' to the address of 'a'. Before, the 7-code patch was working, but it was in the wrong format. Now it is not working altogether.

(I did double check and tried using 7-code patches on a very old build (1.6.0), and the 7-code patches were in fact working as I described previously, in the wrong format, and those same patches are not working on the current build.)

Considering that the issue is now completely different, it might make sense to open a new ticket and close this one. The issue is no longer a minor formatting issue, now it is not working.

**Comment by RedDevilus** [comment]
Or you can just rename the title, and make an addendum for the current issue.

**Comment by rothor** [comment]
Alright, I changed the title. My previous comment has all the details of the current issue.

**Comment by Berylskid** [comment]
All 7-code works flawlessly on current latest nightly build (v1.7.5284).

**Comment by Mrlinkwii** [comment]
@rothor  can you double check :)

**Comment by rothor** [comment]
I can confirm the 7-patch codes are working correctly as of v1.7.5315. I believe this issue can be closed.

I did some research and found [this issue thread](https://github.com/PCSX2/pcsx2/issues/7222) about pnach codes. Apparently pcsx2's patch codes are based on the ps2rd spec, which I didn't know about when I opened this thread. The ps2rd specs can be found [here](https://github.com/mlafeldt/ps2rd/blob/master/Documentation/code_types.txt). Just adding this here to help anyone who might read through this thread.
